src/js/oTable.js

```diff
@@ -131,219 +23,347 @@ function descendingSort(...args) {
-	const intlCollator = getIntlCollator();
-	rows.sort(function (a, b) {
-		let aCol = a.children[index];
-		let bCol = b.children[index];
-
-		if (aCol.getAttribute('data-o-table-order') !== null) {
-			aCol = aCol.getAttribute('data-o-table-order');
-			bCol = bCol.getAttribute('data-o-table-order');
-			if (!isNaN(parseInt(aCol, 10))) {
-				aCol = parseInt(aCol, 10);
-				bCol = parseInt(bCol, 10);
-			}
+function wrapElement(targetEl, wrapEl) {
+	const parentEl = targetEl.parentNode;
+	parentEl.insertBefore(wrapEl, targetEl);
+	wrapEl.appendChild(targetEl);
+}
+
+class OTable {
+	/**
+	 * Initialises an o-table components inside the element passed as the first parameter
+	 *
+	 * @param {(HTMLElement|string)} [el=document.body] - Element where to search for the o-table component. You can pass an HTMLElement or a selector string
+	 * @returns {OTable} - A single OTable instance
+	 */
+	constructor(rootEl) {
+		if (!rootEl) {
+			rootEl = document.body;
+		} else if (!(rootEl instanceof HTMLElement)) {
+			rootEl = document.querySelector(rootEl);
+		}
+		if (rootEl.getAttribute('data-o-component') === "o-table") {
+			this.rootEl = rootEl;
 		} else {
-			aCol = aCol.textContent;
-			bCol = bCol.textContent;
+			this.rootEl = rootEl.querySelector('[data-o-component~="o-table"]');
 		}
 
-		if (isNumericValue) {
-			aCol = parseFloat(aCol.replace(/,/g, ''));
-			bCol = parseFloat(bCol.replace(/,/g, ''));
-		}
+		if (this.rootEl !== undefined) {
+			this.listeners = [];
+			this.isResponsive = false;
+			this.rootEl.setAttribute('data-o-table--js', '');
+
+			// Map "data-o-table-order" to "data-o-table-sort-value".
+			const cellsWithOrder = this.rootEl.querySelectorAll('td[data-o-table-order], th[data-o-table-order]');
+			if (cellsWithOrder.length > 0) {
+				console.warn('o-table: "data-o-table-order" is deprecated, to provide a custom sort value for a table cell use "data-o-table-sort-value" instead.');
+				cellsWithOrder.forEach(cell => {
+					if (cell.getAttribute('data-o-table-order') !== null) {
+						cell.setAttribute('data-o-table-sort-value', cell.getAttribute('data-o-table-order'));
+					}
+				});
+			}
 
-		if (sortAscending) {
-			return ascendingSort(aCol, bCol, isNumericValue, intlCollator);
-		} else {
-			return descendingSort(aCol, bCol, isNumericValue, intlCollator);
+			this.tableHeaders = Array.from(this.rootEl.querySelectorAll('thead th'));
+			const tableRows = Array.from(this.rootEl.getElementsByTagName('tr'));
+
+			this.tableHeaders.forEach((th, columnIndex) => {
+				// Do not sort headers with attribute.
+				if (th.hasAttribute('data-o-table-heading-disable-sort')) {
+					return false;
+				}
+
+				th.setAttribute('tabindex', "0");
+
+				const listener = this._sortByColumn(columnIndex);
+				this.listeners.push(listener);
+				th.addEventListener('click', listener);
+				th.addEventListener('keydown', (event) => {
+					const ENTER = 13;
+					const SPACE = 32;
+					if ('code' in event) {
```

Comment:
Should we use [`key`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) instead of `code`?

Reply:
I'm not sure that matters here? For ENTER and SPACE, we don't care about modifier keys such as shift?